### ISSUE TITLE
Multilevel models not working in develop version

### ISSUE THREAD

**Issue by andymilne** [stan-dev/projpred#55]
Very likely I am doing something wrong – I installed the developer version with `devtools::install_github('stan-dev/projpred', build_vignettes = TRUE, force = TRUE)`.

I have tried a multilevel model fitted in brms and also in rstanarm. In both cases, when running `varsel`, I get an error saying that multilevel models are not supported:

`Error in get_refmodel.stanreg(object, ...) : 
  stan_lmer and stan_glmer are not yet supported.` and
```
Error in get_refmodel.brmsfit(object, ...) : 
  Multilevel or other special terms are not yet supported.
In addition: Warning message:
Method 'parse_bf' is deprecated. Please use 'brmsterms' instead. 
```

**Comment by AlejandroCatalina** [comment]
Indeed that’s installing only the master branch from github. You need to pass a second argument “develop” to install_github. Beware that it’s still under development! Thanks for your interest tho.

**Comment by andymilne** [comment]
Thank you, that works!

**Comment by andymilne** [comment]
Just a comment really -- I have managed to run varsel on a multilevel model and done a projection onto the recommended subset of coefficients. The coefficients chosen and their effects in the projected model are very surprising (I do not believe they are an appropriate subset) and many of them go in unexpected directions (i.e., different to what I get if I run any reasonable subset of the model fresh), and it's not clear why the projected random effects can take negative values (shouldn't these be standard deviations?). There didn't seem to be any warnings or errors during the varsel run, other than lmer complaints about singular fits, which I assume can be ignored.

In summary, it is quite possible that I am completely misunderstanding how this should work, but I was surprised by the results. For now, I will go back to variable selection with single-level models where the results make sense.

**Comment by AlejandroCatalina** [comment]
Two comments regarding your experience:

1. Singular fits can **sometimes** be ignored. If the model is gaussian you can be fairly confident they're not an issue. If the model is not gaussian, well, there can be undefined behaviour. The underlying solver is `lme4`, and what generally happens is that the model is not stable for a subset of covariates. We are working on improving this for future versions as with multilevel models this seems to be a pretty common case.

**Comment by andymilne** [comment]
Thanks for the comments -- this is a Gaussian model, but I will try the `ndraws_pred` and `ndraws` arguments to see if that helps. I would be happy to supply the data and model formula if that would be of interest but it sounds like you are already familiar with potential issues (although because this is real-world data there is, of course, no ground truth).

**Comment by AlejandroCatalina** [comment]
I would gladly take a look at the model! We have found that Gaussian models usually behave well in terms of predictions (the ELPD matches and so on) but the projected posterior can very well be different for some effects. One potential issue is that in some multilevel models the overall effects can be soaked up into the group intercepts, and that can greatly affect the posterior. The predictions would still make sense, but we would lose identifiability on the overall effects. That's something we are also looking into!

**Comment by andymilne** [comment]
Thanks Alejandro -- I will upload somewhere and email you a link. But please only spend time on this if it is useful to you/projpred because I have alternative strategies for simplifying my model. 

The data and model here represents a case where there are a number of predictors with some very high multicollinearities and a maximal random effects structure, so may be particularly challenging.

**Comment by AlejandroCatalina** [comment]
That seems a complex and challenging model indeed. Having a new example will help me to better understand what's going on, thanks!
